### PR TITLE
add data rate calculations

### PR DESCRIPTION
**Pull request by davidk81**
- added `--stats` flag which enables statistics output:
  - accumulated messages received
  - accumulated bytes received
  - average data rate
  - start time is calculated using time of first telemetry message receive, this is technically not accurate but will converge over time.
- created throttled_logger 
  - allows for rate-based logging (throttled messages are dropped)
  - allows for convenient debug/verbose logging
  - allows for single-line overwrite mode (used for stats logging)
- added shutdown handler to log stats on exit, this is meaningful because some stats may not be logged due to log throttling
- replaced default "log" with custom logger to address new-line issue when the ovelay `--stats` feature is turned on

### CODE REVIEW

**Comment by davidk81** [comment]
example output
image

**Comment by davidk81** [comment]
- stats for current plan only logs to same line
- when plan changes, new line is created
- instantaneous rate & delay samples most recent 5 seconds of data (but no less than 5 data samples)
image

File: cmd/flag/stats_flag.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
+//
+// Copyright © 2019 Infostellar, Inc.
```
Comment (danada) [inline cmd/flag/stats_flag.go]: License year keeps going further back in time in lower files 😆

**Comment by davidk81** [comment]
image

File: pkg/satellite/plan/list_plans.go [hunk 2]
```diff
@@ -16,13 +16,14 @@ package plan
...
 	"context"
-	"log"
 	"time"
 
 	"github.com/golang/protobuf/ptypes"
 
 	stellarstation "github.com/infostellarinc/go-stellarstation/api/v1"
 	"github.com/infostellarinc/stellarcli/pkg/apiclient"
+	log "github.com/infostellarinc/stellarcli/pkg/logger"
+
```
Comment (danada) [inline pkg/satellite/plan/list_plans.go]: Nit: newline

File: pkg/satellite/tle/set_tle_source.go [hunk 3]
```diff
@@ -17,9 +17,10 @@ package tle
 import (
 	"context"
 	"fmt"
-	"log"
 	"strings"
 
+	log "github.com/infostellarinc/stellarcli/pkg/logger"
+
```
Comment (danada) [inline pkg/satellite/tle/set_tle_source.go]: Nit: newline (and once grouped will probably have to be reordered)

**Review by danada — APPROVED** [review]
Thanks this is looking really good.

**Pull request merged**